COMPONENT-GUIDELINES.md

```diff
@@ -0,0 +1,293 @@
+# Purpose
```

Comment:
Small things I think could be worth adding:

- If the app has a well-supported Node.js client library, that should be preferred to HTTP requests for a few reasons (reduces code / maintenance, handles retries, etc.)
- If the user is asked to enter sensitive data in props, make sure it's a secret prop.
- You called out pagination for async options, but we should also make sure users support pagination when retrieving _all_ data (e.g. event data for polling sources).

Reply:
good points. done